File: rafael/controle-de-fluxo/EscolheCaminho.java
class EscolheCaminho {
	public static void main(String[] args) {
		double valor = Math.random();

		System.out.println("VALOR: " + valor);

		if (valor < 0.5) {
			System.out.println("Vire à esquerda");

			valor = Math.random();
	
			System.out.println("VALOR: " + valor);
	
			if (valor < 0.5) {
				System.out.println("Vire à esquerda");
			} else {
				System.out.println("Vire à direita");
			}
	
		} else {
			System.out.println("Vire à direita");

			valor = Math.random();

			System.out.println("VALOR: " + valor);
	
			if (valor < 0.5) {
				System.out.println("Vire à esquerda");
			} else {
				System.out.println("Vire à direita");
			}
		}
	}
}

File: rafael/controle-de-fluxo/EscolheCaminho.cs
class EscolheCaminho 
{
	static void Main() 
	{
		System.Random gerador = new System.Random();

		double valor = gerador.NextDouble();

		System.Console.WriteLine("VALOR: " + valor);

		if (valor < 0.5) 
		{
			System.Console.WriteLine("Vire à esquerda");

			valor = gerador.NextDouble();
	
			System.Console.WriteLine("VALOR: " + valor);
	
			if (valor < 0.5) 
			{
				System.Console.WriteLine("Vire à esquerda");
			} 
			else 
			{
				System.Console.WriteLine("Vire à direita");
			}
	
		} 
		else 
		{
			System.Console.WriteLine("Vire à direita");

			valor = gerador.NextDouble();

			System.Console.WriteLine("VALOR: " + valor);
	
			if (valor < 0.5) 
			{
				System.Console.WriteLine("Vire à esquerda");
			} 
			else 
			{
				System.Console.WriteLine("Vire à direita");
			}
		}
	}
}
